src/OpenTelemetry.AutoInstrumentation/Instrumentations/NoCode/NoCodeIntegrationHelper.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Reflection;
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+using OpenTelemetry.AutoInstrumentation.Configurations;
+using OpenTelemetry.AutoInstrumentation.Util;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.NoCode;
+
+internal static class NoCodeIntegrationHelper
+{
+    private static readonly ActivitySource Source = new("OpenTelemetry.AutoInstrumentation.NoCode");
+
+    internal static List<NoCodeInstrumentedMethod> NoCodeEntries { get; set; } = [];
+
+    internal static CallTargetState OnMethodBegin()
+    {
+        const int methodNameFrameIndex = 3;
+
+        var method = new StackFrame(methodNameFrameIndex).GetMethod();
+        var methodName = method?.Name;
+        var typeName = method?.DeclaringType?.FullName;
+        var assemblyName = method?.DeclaringType?.Assembly.GetName().Name;
+        var parameters = method?.GetParameters()!;
+        var noCodeEntry = NoCodeEntries.Single(x =>
+            x.Definition.TargetType == typeName &&
+            x.Definition.TargetMethod == methodName &&
+            x.Definition.TargetAssembly == assemblyName &&
+            CheckParameters(x.SignatureTypes, parameters));
+
+        // TODO Span kind and attributes from configuration
+        // TODO Consider execute some dynamic code to build name/span/attributes based on config and taking data from method parameters
+        var activity = Source.StartActivity(name: noCodeEntry.SpanName, kind: ActivityKind.Internal, tags: [new("TODO", "attributes-from-configuration")]);
```

Comment:
Is this `TODO` tag a carry-over from testing?

Reply:
Nope, intentionally left it here, as the comments 2 lines before.
I have a plan to add this 2 missing functionalities in the follow up, if you think that we need to remove this TODO from the code, I also can do this.